### PR TITLE
PLAT-1290-3 StoredFileConverter: Convert images to PDF via ImageIO

### PR DESCRIPTION
**Pull request by vdcstr**

### CODE REVIEW

File: platform-common/src/main/java/com/softicar/platform/common/io/mime/MimeType.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
  */
 public enum MimeType implements IMimeType {
 
-	APPLICATION_JSON("application/json"),
+	APPLICATION_JSON("application/json", "json"),
```
Comment (alexander-schmidt) [inline platform-common/src/main/java/com/softicar/platform/common/io/mime/MimeType.java:16]: Unrelated fix; sneeking this in here.

File: platform-common/src/main/java/com/softicar/platform/common/ui/image/Images.java [hunk 2]
```diff
@@ -129,4 +130,34 @@ public static long countPixelsWithColor(BufferedImage image, Color expectedColor
...
+	/**
+	 * Determines whether {@link #readImages(Supplier)} can handle images with
+	 * the given {@link IMimeType}.
+	 *
+	 * @param mimeType
+	 *            the image {@link IMimeType} (never <i>null</i>)
+	 * @return <i>true</i> if the given image {@link IMimeType} is supported;
+	 *         <i>false</i> otherwise
+	 */
+	public static boolean isSupported(IMimeType mimeType) {
```
Comment (oliver-richers) [inline platform-common/src/main/java/com/softicar/platform/common/ui/image/Images.java:143]: since this only tests for image readers, maybe the name should be `isReadable` or `canRead` or `hasReader`, you get the point :)

File: platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java [hunk 3]
```diff
@@ -62,6 +63,35 @@ public void testCountPixelsWithColor() {
 		assertEquals(3067, Images.countPixelsWithColor(image, textColor));
 	}
 
+	@Test
+	public void testIsSupportedWithMimeTypeJpg() {
```
Comment (oliver-richers) [inline platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java:67]: why is only `Jpg` in the test method name?

File: platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java [hunk 4]
```diff
@@ -62,6 +63,35 @@ public void testCountPixelsWithColor() {
...
+
+		assertTrue(Images.isSupported(MimeType.IMAGE_JPEG));
+		assertTrue(Images.isSupported(MimeType.IMAGE_PNG));
+		assertTrue(Images.isSupported(MimeType.IMAGE_TIFF));
+		assertFalse(Images.isSupported(MimeType.APPLICATION_PDF));
+	}
+
+	@Test
+	public void testIsSupportedWithSuffixJpg() {
```
Comment (oliver-richers) [inline platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java:76]: why is only `Jpg` in the test method name?

File: platform-ajax/src/test/java/com/softicar/platform/ajax/AjaxTestResources.java [hunk 5]
```diff
@@ -16,6 +16,6 @@ public interface AjaxTestResources {
 
 	IResourceSupplierFactory FACTORY = new DomResourceSupplierProxyFactory(AjaxTestResources.class);
 
-	IResourceSupplier TEST_PNG = FACTORY.create("test.png");
-	IResourceSupplier TEST_TIFF = FACTORY.create("test.tiff");
+	IResourceSupplier PNG_TEST = FACTORY.create("png-test.png");
```
Comment (alexander-schmidt) [inline platform-ajax/src/test/java/com/softicar/platform/ajax/AjaxTestResources.java:19]: This renaming was necessary because otherwise the resources enumerated here had colliding keys, which in turn resulted in a pseudo-(non-)determinism in `AjaxDomEngineTest`.

File: platform-common-pdf/src/test/java/com/softicar/platform/common/pdf/PdfTestFiles.java [hunk 6]
```diff
@@ -12,5 +12,5 @@ public interface PdfTestFiles {
 
 	IResourceSupplier PDF_WITH_BLANK_PAGES = FACTORY.create("pdf-with-blank-pages.pdf");
 	IResourceSupplier PDF_WITHOUT_BLANK_PAGES = FACTORY.create("pdf-without-blank-pages.pdf");
-	IResourceSupplier TIF_MULTI_IMAGE = FACTORY.create("tif-multi-image.tif");
+	IResourceSupplier TIFF_MULTI_IMAGE = FACTORY.create("tiff-multi-image.tiff");
```
Comment (alexander-schmidt) [inline platform-common-pdf/src/test/java/com/softicar/platform/common/pdf/PdfTestFiles.java:15]: Renamed various tif(f) resources to correspond to the name of the mime type (`image/tiff`), and because `tiff` is far more widespread than `tif`.

File: platform-ajax/src/test/java/com/softicar/platform/ajax/AjaxTestResources.java (hunk 5 above)
Comment (oliver-richers) [reply platform-ajax/src/test/java/com/softicar/platform/ajax/AjaxTestResources.java:19]: nice

File: platform-common/src/main/java/com/softicar/platform/common/io/mime/MimeType.java (hunk 1 above)
Comment (oliver-richers) [reply platform-common/src/main/java/com/softicar/platform/common/io/mime/MimeType.java:16]: nice

**Review by oliver-richers — APPROVED** [review]

File: platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java (hunk 3 above)
Comment (alexander-schmidt) [reply platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java:67]: Oops. Artifact.

File: platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java (hunk 4 above)
Comment (alexander-schmidt) [reply platform-common/src/test/java/com/softicar/platform/common/ui/image/ImagesTest.java:76]: Oops. Artifact.

**Review by oliver-richers — APPROVED** [review]
